src/test/java/com/epam/training/circularbuffer/CircularBufferTest.java

```diff
@@ -0,0 +1,121 @@
+package com.epam.training.circularbuffer;
+
+import com.epam.training.exception.NotEnoughSpaceException;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class CircularBufferTest {
+    private CircularBuffer<Integer> testingInstance;
+
+    @Before
+    public void setUp() {
+        testingInstance = new CircularBuffer<>(5);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void shouldNotPutInTheFullArray() {
+        testingInstance.put(1);
+        testingInstance.put(1);
+        testingInstance.put(1);
+        testingInstance.put(1);
+        testingInstance.put(1);
+        testingInstance.put(1);
+    }
+
+    @Test
+    public void shouldGetTheTail() {
+        testingInstance.put(1);
+        testingInstance.put(2);
+        assertEquals(1, (int) testingInstance.get());
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void shouldNotGetFromEmptyBuffer() {
+        testingInstance.get();
+    }
+
+    @Test
+    public void shouldConvertToObjectArray() {
+        testingInstance.put(1);
+        testingInstance.put(2);
+        testingInstance.put(3);
+        testingInstance.get();
+        Object[] actual = testingInstance.toObjectArray();
+
+        assertEquals(2, actual.length);
+        assertEquals(2, actual[0]);
+        assertEquals(3, actual[1]);
+    }
+
+    @Ignore
```

Comment:
Do not commit ignored tests in repo.

Reply:
It's for you to show the method I can't implement.
Maybe, you could help me with the implementation or delete this method at all.